#1652 Testing iOS Libraries with objection (#1789)
OWASP/owasp-mstg: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

It need to be validated whether the copyrights of the licenses have been adhered to. This often means that the application should have an `about` or `EULA` section in which the copy-right statements are noted as required by the license of the third party library.

> ##### Listing Application Libraries

Reviewer: Please commit this suggestion after merging the current `master` which has reduced one level on all titles.
```suggestion
#### Listing Application Libraries
```

It need to be validated whether the copyrights of the licenses have been adhered to. This often means that the application should have an `about` or `EULA` section in which the copy-right statements are noted as required by the license of the third party library.

##### Listing Application Libraries

> When performing app analysis, it is important to also analyze the app dependencies (usually in form of libraries or so-called iOS Frameworks) and ensure that they don't contain any vulnerabilities. Even when you don't have the source code, you can still identify some of the app dependiencies using tools like [objection](https://github.com/sensepost/objection), [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF) or otool. Objection is the recommended tool, it gives the most accurate results and it is easy to use.

Reviewer: Corrected word: dependencies

Merged below sentence here. I'd remove then lines 326-329 since we don't need a subsection anymore.

```suggestion
When performing app analysis, it is important to also analyze the app dependencies (usually in form of libraries or so-called iOS Frameworks) and ensure that they don't contain any vulnerabilities. Even when you don't have the source code, you can still identify some of the app dependencies using tools like [objection](https://github.com/sensepost/objection), [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF) or otool. Objection is the recommended tool, since it provides the most accurate results and it is easy to use. It contains a module to work with iOS Bundles, which offers two commands: `list_bundles` and `list_frameworks`.
```

> 
> ###### objection
> 
> Objection contains a module to work with iOS Bundles, which offers two commands: `list_bundles` and `list_frameworks`.

Reviewer: As indicated in the other suggestion.
```suggestion
```

It need to be validated whether the copyrights of the licenses have been adhered to. This often means that the application should have an `about` or `EULA` section in which the copy-right statements are noted as required by the license of the third party library.

##### Listing application libraries

When no source-code is available for libraries analysis, you can find some of the frameworks being used with [objection](https://github.com/sensepost/objection), [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF) or otool.

###### objection

> objection contains bundles module to work with iOS Bundles. This module has two commands `list_bundles` and `list_frameworks`.

Author: Corrected to capitalised letter, I think it is correct iOS bundles See: https://developer.apple.com/app-store/app-bundles/

`list_bundles` command lists all of the application’s non framework bundles. The output of both commands contains executable name, bundle id, version of the library and path to the library.

```bash
...itudehacks.DVIAswiftv2.develop on (iPhone: 13.2.3) [usb] # ios bundles list_bundles
Executable    Bundle                                       Version  Path
------------  -----------------------------------------  ---------  -------------------------------------------
DVIA-v2       com.highaltitudehacks.DVIAswiftv2.develop          2  ...-1F0C-4DB1-8C39-04ACBFFEE7C8/DVIA-v2.app
CoreGlyphs    com.apple.CoreGlyphs                               1  ...m/Library/CoreServices/CoreGlyphs.bundle
```

`list_frameworks` command lists all of the application’s bundles that represent frameworks.

> 

Author: Version might not be available since those frameworks are being provided by Apple, Other frameworks have versions.
But I can also spot apple bundle with the version, so it may also be objection bug.
`RawCamera     com.apple.RawCamera.bundle                 9.00.1     ...em/Library/CoreServices/RawCamera.bundle`

Parse           org.cocoapods.Parse                        1.17.3     ...8/DVIA-v2.app/Frameworks/Parse.framework
                                                                      ...rivateFrameworks/ASEProcessing.framework
                                                                      ...teFrameworks/AppleNeuralEngine.framework
DVIA-v2         com.highaltitudehacks.DVIAswiftv2.develop  2.0        ...-1F0C-4DB1-8C39-04ACBFFEE7C8/DVIA-v2.app
                                                                      ...rameworks/CoreLocationProtobuf.framework
Flurry_iOS_SDK  org.cocoapods.Flurry-iOS-SDK               10.0.2     ....app/Frameworks/Flurry_iOS_SDK.framework
RealmSwift      org.cocoapods.RealmSwift                   4.1.1      ...A-v2.app/Frameworks/RealmSwift.framework
Realm           org.cocoapods.Realm                        4.1.1      ...8/DVIA-v2.app/Frameworks/Realm.framework
                                                                      ...vateFrameworks/CoreDuetContext.framework
                                                                      ...ystem/Library/Frameworks/IOKit.framework
```

###### MobSF

> After obtaining the application ipa file, it can be uploaded to the MobSF. MobSF lists external and system libraries, however it provides wrong versions of some libraries. This can be seen by comparing objection results with mobsf results.

Reviewer 2: > however it provides wrong versions of some libraries

I would slightly rephrase this, and possibly place it as its own sentence. "However, it occassionally provides the incorrect version of some libraries."

> results with mobsf results

Perhaps capitalize this consistently: MobSF.

Author: Corrected, thanks for the comments :)
